tests/test_runner.py

```diff
@@ -149,3 +162,32 @@ def test_merging(example_input_cfg: dict, tmp_path: Path, monkeypatch: pytest.Mo
         # demo version of SHIELD-HIT12A cannot use different random seeds for each run,
         # therefore the first bin dose should be the same for all runs
         assert first_bin_dose == data_first_bin_dose
+
+
+@pytest.mark.smoke
+@pytest.mark.skipif(sys.platform == "darwin", reason="we don't have SHIELD-HIT12A demo binary for MacOSX")
+@pytest.mark.skipif(sys.platform == "win32", reason="simulator mocks don't work on Windows")
+def test_topas(topas_mock_path: Path, topas_input_path: Path, tmp_path: Path):
+    """Test if runner can run topas mock and read the output"""
+    settings = SimulationSettings(input_path=str(topas_input_path),
+                                    simulator_exec_path=str(topas_mock_path))
+    logging.info(settings)
+
+    r = Runner(settings=settings, jobs=2, output_directory=tmp_path)
+    
+    isRunOk = r.run()
+    assert isRunOk
+    
+    #ensure that correct number of threads is set in the input file
+    with open(r.settings.input_path, "r") as input_file:
+        contents = input_file.read()
+        assert "i:Ts/NumberOfThreads = 0" not in contents
+        assert "i:Ts/NumberOfThreads = 2" in contents
+        
+    data = r.get_data()
+    logging.info(data)
+    assert data is not None
+    assert 'fluence_bp_protons_xy' in data
+    assert 'fluence_bp_protons_xy2' in data
+    
+    shutil.rmtree(tmp_path)
```

Comment:
```suggestion
```

Its bad practice to remove tmp_path. pytest always keep couple of recent directories and removes oldest using round robin algorithm.  See https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html

https://docs.pytest.org/en/7.1.x/reference/reference.html#tmp-path

![image](https://github.com/DataMedSci/pymchelper/assets/7374733/f863178c-25b0-4903-ba9b-a7161866e44a)